pkg/apis/crd/v1alpha1/types.go

```diff
@@ -591,3 +596,17 @@ type TierList struct {
 
 	Items []Tier `json:"items"`
 }
+
+type ServiceAccount struct {
+	// Name and Namespace must be used together to select a specific ServiceAccount.
+	// Name and Namespace cannot be set with any other fields.
+	Name      string `json:"name,omitempty"`
+	Namespace string `json:"namespace,omitempty"`
+	// Select all ServiceAccounts matched by this selector. If set with
+	// NamespaceSelector, ServiceAccounts are matched from Namespaces matched by the
+	// NamespaceSelector.
+	// Cannot be set with any other fields except NamespaceSelector.
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+	// Can be only used with LabelSelector.
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
```

Comment:
Oh, sorry I misunderstood the discussion. 
After thinking about the difference between those two designs, I think from the development perspective, using its own `NamespaceSelector` could make implementation easier. `serviceAccounts` will be a stand-alone field. And from the API perspective, seems they are both clear. 
For now, I came up with two use cases that these two designs would be different:
1. We want to use name+namespace and labelSelector to select ServiceAccount at the same time.
```
# Use its own nsSel
- serviceAccounts:
  - labelSelector:
      matchLabels:
        level: admin
    namespaceSelector:
      matchLabels:
        env: prod
  - name: sa-1
    namespace: ns-1

# Use exist nsSel
- namespaceSelector:
      matchLabels:
        env: prod 
  serviceAccounts:
    - labelSelector:
        matchLabels:
          level: admin
- serviceAccounts:
  - name: sa-1
    namespace: ns-1
```
2. We want to select multi-labels ServiceAccount from the same nsSel
```
# Use its own nsSel
- serviceAccounts:
  - labelSelector:
      matchLabels:
        level: admin
    namespaceSelector:
      matchLabels:
        env: prod
  - labelSelector:
      matchLabels:
        level: user
    namespaceSelector:
      matchLabels:
        env: prod

# Use exist nsSel
- namespaceSelector:
      matchLabels:
        env: prod 
  serviceAccounts:
    - labelSelector:
        matchLabels:
          level: admin
    - labelSelector:
        matchLabels:
          level: user
```
Glad to hear your ideas about which one is better. Thanks.

Reply:
cc @tnqn @jianjuns @Dyanngg

Reply:
I guess the confusion here is from we support using both name and labelSelector to select ServiceAccounts. If we support only name, then it is quite clear ServiceAccount is self-complete. In this sense, I slightly prefer to add a new NamespaceSelector into ServiceAccount.

And I start to think is it really needed to support labelSelector to select ServiceAccounts. Do we know other solutions support that?

Reply:
Thanks for sharing your thoughts. As Antonin mentioned [here](https://github.com/antrea-io/antrea/issues/2927#issuecomment-949850005), based on Calico's doc, they support label selector.

Reply:
A fresh learning from Tim Hockin during the meeting we had yesterday for K8s upstream CNP discussions:
It seems that from K8s upstream perspective, people are leaning more towards extremely verbose, yet more explicit selector structs. Specifically, for this PR an API design preferred may look like:
```
- serviceAccounts:
   - namespacedName:
        namespace: test
        name: test-serviceAccount 
   - namespaceLabelAndServiceAccountLabelSelector:
        namespaceLabelSelector:
           matchLabels:
                env: prod
        serviceAccountLabelSelector:
            matchLabels:
                level: user
```
TL;DR Obviously there is more redundancy in the spec compared to what we have right now, but there are some major benefits, the most important one is being able to fail-close as the API evolves. How that would work is:
- From the openAPI validation, we ensure that only oneOf the `namespacedName` or `namespaceLabelAndServiceAccountLabelSelector` selection type is used for a single `serviceAccount` entry. We can more selection methods e.g. `namespaceNameAndServiceAccountLabelSelector`, but the important thing is that we explicit list out any selector combinations that are valid, and provide each such selectors as a field of that dedicated selector combination type. This also means that we might not even need admission control validations.
- From the controller, we ensure that for any serviceAccount specified in an ACNP, we can understand it as a `namespacedName` or `namespaceLabelAndServiceAccountLabelSelector` or any other combination we support. Otherwise, we immediately know that there must be some sort of version skew between the ACNP object and Antrea Controller and act accordingly, instead of interpreting that entry in a wrong way.

Reply:
@Dyanngg Thanks for the proposal. It sounds good to me. I have pushed a version following your ideas. Hope you can take a look.

Reply:
What about
```
- serviceAccounts: 
  - namespacedServiceAccountSelector:
      namespaceSelector: xxx
      serviceAccountSelector: yyy
```

Reply:
> And I start to think is it really needed to support labelSelector to select ServiceAccounts. Do we know other solutions support that?

Perhaps too late to bring this up but looking at the cost of supporting ServiceAccount labelSelector I would like to share my thoughts:

For the original issue https://github.com/antrea-io/antrea/issues/2927 and common use cases, it's straightforward to use single service account for applications having same permission and security requirement, instead of creating multiple service accounts with same labels. If there is a scenario that multiple applications using different service accounts but wanting to have same security requirements,  labeling Pods directly is still more convinient and straightforward than labeling service accounts as @antoninbas commented in https://github.com/antrea-io/antrea/issues/2927#issuecomment-949850005.

And unlike Pods which are created and destoryed dynamically and names are unpredictable, Service and ServiceAccount are relatively static and are created on purpose with meaningful names. There shouldn't be multiple Services and ServiceAccounts created for same purposes, it should be easy to enumerate their names.

From implementation perspective, it's a bit complicated and heavy to implement serviceAccount LabelSelector, which is essentially selector of selector, like Service LabelSelector. I recall we wanted to support Service LabelSelector as well and aborted because of similar concern, and I haven't heard anyone complain Service name cannot meet their requirement.

Before we have real use cases, I feel not worth to support serviceAccount LabelSelector with the cost of implementing "selector of selector".

Reply:
Yes, I would second Quan's points. If supporting labelSelector is much complexer, I would suggest to start from supporting only names too.

Reply:
Make sense. And this could also avoid this feature being blocked by the verbose API design discussion.